tests/unit/maxmemory.tcl

```diff
@@ -1,3 +1,141 @@
+start_server {tags {"maxmemory" "external:skip"}} {
+    r config set maxmemory 10mb
+    r config set maxmemory-policy allkeys-lru
+    set server_pid [s process_id]
+
+    proc init_test {client_eviction} {
+        r flushdb
+
+        set prev_maxmemory_clients [r config get maxmemory-clients]
+        if $client_eviction {
+            r config set maxmemory-clients 3mb
+        } else {
+            r config set maxmemory-clients 0
+        }
+
+        r config resetstat
+        # fill 5mb using 50 keys of 100kb
+        for {set j 0} {$j < 50} {incr j} {
+            r setrange $j 100000 x
+        }
+        assert_equal [r dbsize] 50
+    }
+    
+    proc verify_test {client_eviction} {
+        set evicted_keys [s evicted_keys]
+        set evicted_clients [s evicted_clients]
+        set dbsize [r dbsize]
+        
+        if $::verbose {
+            puts "evicted keys: $evicted_keys"
+            puts "evicted clients: $evicted_clients"
+            puts "dbsize: $dbsize"
+        }
+
+        if $client_eviction {
+            assert_morethan $evicted_clients 0
+            assert_equal $evicted_keys 0
+            assert_equal $dbsize 50
+        } else {
+            assert {$evicted_clients == 0 && $evicted_keys > 0 && $dbsize < 50}
+            assert_equal $evicted_clients 0
+            assert_morethan $evicted_keys 0
+            assert_lessthan $dbsize 50
+        }
+    }
+
+    foreach {client_eviction} {false true} {
+        set clients {}
+        test "eviction due to output buffers of many MGET clients, client eviction: $client_eviction" {
+            init_test $client_eviction
+
+            for {set j 0} {$j < 20} {incr j} {
+                set rr [redis_deferring_client]
+                lappend clients $rr
+            }
+
+            # Freeze the server so output buffers will be filled in one event loop when we un-freeze after sending mgets
+            exec kill -SIGSTOP $server_pid
+            for {set j 0} {$j < 5} {incr j} {
+                foreach rr $clients {
+                    $rr mget 1
```

Comment:
this is not a very realistic test (mget of one huge key). i assume it was the other way around and was changed in order the make the test predictable?

Reply:
I think this could have been a `GET` command and doesn't need to be `MGET`. The idea is to have many clients with big output buffers. 
What do you mean "not a very realistic tests"?

Reply:
the typical case of `KEYS *` like scenario that cases mass eviction is a burst of MGETs filling the output buffer.
i think this is typically a lot of MGET calls on many key by many clients.
so i just commented that IIRC when this test was initially created (added the tests before the code to fix them), it was a more typical case, and it was change to be something that's easier to test (considering we know the actual problem can be triggered by one command on a single key too).

just a random comment, we can dismiss it.